### PR TITLE
[WIP] Extended metadata testing with Pulsar 

### PR DESCRIPTION
**Pull request by jmchilton**

### CODE REVIEW

File: lib/galaxy_test/driver/driver_util.py [hunk 1]
```diff
@@ -517,8 +517,8 @@ def serve_webapp(webapp, port=None, host=None):
         try:
             server = httpserver.serve(webapp, host=host, port=port, start_loop=False)
             break
-        except socket.error as e:
-            if e[0] == 98:
+        except OSError as e:
```
Comment (nsoranzo) [inline lib/galaxy_test/driver/driver_util.py]: If you are interested in keeping Python 2.7 compatibility, this should work on 2.7 and 3.5-3.8:

```suggestion
        except socket.error as e:
```

**Comment by jmchilton** [comment]
Closing to not re-run Jenkins tests each time.

**Comment by jmchilton** [comment]
Potentially useful breadcrumb for future @jmchilton  https://github.com/galaxyproject/galaxy/commit/067b76308aa648068ce8e9e4d5f6d9be199dbe3a

**Comment by jmchilton** [comment]
Got it submitted jobs in my branch and I've got some debugging working, not sure how to reference a service running in the test container from the job container in Kubernetes.

```
2020-02-26T17:20:40.025336168Z 2020-02-26 17:20:40,020 INFO  [pulsar.client.staging.down][[manager=_default_]-[action=postprocess]-[job=1]] collecting output None with action FileAction[url=http://host.docker.internal:9258/api/jobs/adb5f5c93f827949/files?job_key=56a9d5119a62bf92&path=/tmp/tmpdcxdbyhs/d/5/5/dataset_d558a172-e259-4a57-b170-5797e4905b43.dat&file_type=output,action_type=remote_transfer,path=/tmp/tmpdcxdbyhs/d/5/5/dataset_d558a172-e259-4a57-b170-5797e4905b43.dat]
2020-02-26T17:20:40.115809592Z 2020-02-26 17:20:40,114 ERROR [pulsar.client.transport.poster][[manager=_default_]-[action=postprocess]-[job=1]] Problem with poster post of [/pulsar_staging/1/outputs/dataset_d558a172-e259-4a57-b170-5797e4905b43.dat]
2020-02-26T17:20:40.115830493Z Traceback (most recent call last):
2020-02-26T17:20:40.115834893Z   File "/usr/local/lib/python2.7/site-packages/pulsar/client/transport/poster.py", line 33, in post_file
2020-02-26T17:20:40.115839293Z     return urlopen(request).read()
2020-02-26T17:20:40.115843093Z   File "/usr/local/lib/python2.7/urllib2.py", line 154, in urlopen
2020-02-26T17:20:40.115847093Z     return opener.open(url, data, timeout)
2020-02-26T17:20:40.115850593Z   File "/usr/local/lib/python2.7/urllib2.py", line 429, in open
2020-02-26T17:20:40.115866694Z     response = self._open(req, data)
2020-02-26T17:20:40.115870294Z   File "/usr/local/lib/python2.7/urllib2.py", line 447, in _open
2020-02-26T17:20:40.115873494Z     '_open', req)
2020-02-26T17:20:40.115876394Z   File "/usr/local/lib/python2.7/urllib2.py", line 407, in _call_chain
2020-02-26T17:20:40.115879594Z     result = func(*args)
2020-02-26T17:20:40.115882594Z   File "/usr/local/lib/python2.7/site-packages/poster/streaminghttp.py", line 142, in http_open
2020-02-26T17:20:40.115885694Z     return self.do_open(StreamingHTTPConnection, req)
2020-02-26T17:20:40.115888794Z   File "/usr/local/lib/python2.7/urllib2.py", line 1198, in do_open
2020-02-26T17:20:40.115891994Z     raise URLError(err)
2020-02-26T17:20:40.115894895Z URLError: <urlopen error [Errno -2] Name or service not known>
```

**Comment by jmchilton** [comment]
@mvdbeek quick question - feel free to ignore if you don't have any clues, any guess for what http://host.docker.internal:9258 should be instead for Github Actions? I have no clue how to to reference that container the tests are running inside of.

**Comment by mvdbeek** [comment]
I don't think we can do this on github actions (using minikube). I think this would work if we deployed Galaxy in kubernetes though, since then you can address it in there.

**Comment by jmchilton** [comment]
Actually it seems to be hitting the job files API fine from inside Kubernetes if I set uwsgi to serve on 0.0.0.0 and if I use socket.gethostname() as the infrastructure URL. The test is still failing and I don't know why but I think the communication is possible.

https://github.com/jmchilton/galaxy/runs/470709177?check_suite_focus=true

Which is good I think it is needed for this work on remote metadata and for the work on ITs #9448. Having to deploy Galaxy in Kubernetes to use it would be pretty heavy.

**Pull request closed without merging**